aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -2147,6 +2147,53 @@ Tensor& take_along_dim_out(const Tensor& self, const Tensor& indices, std::optio
   return at::gather_out(result, self.view(-1), 0, indices.view(-1));
 }
 
+namespace {
+
+inline std::tuple<Tensor, Tensor> _put_along_dim_helper(const Tensor& self, const Tensor& indices, const Tensor& values, int64_t dim) {
+  auto new_indices_sizes = DimVector(indices.sizes());
+  for (int i = 0; i < indices.dim(); i++) {
+    if (i == dim) {
+      new_indices_sizes[i] = indices.size(i);
+    } else {
+      new_indices_sizes[i] = self.size(i);
+    }
+  }
+  auto new_indices = at::broadcast_to(indices, new_indices_sizes);
+  auto new_values = at::broadcast_to(values, new_indices_sizes);
+  return std::make_tuple(new_indices, new_values);
+}
+}
+
+Tensor put_along_dim(
+    const Tensor& self,
+    const Tensor& indices,
+    const Tensor& values,
+    c10::optional<int64_t> opt_dim) {
+  checkDevice("torch.put_along_dim():", {self, indices, values}, self.device());
+  if (opt_dim.has_value()) {
+    TORCH_CHECK(opt_dim.has_value(), "None dim is not supported.");
+    auto [new_indices, new_values] = _put_along_dim_helper(self, indices, values, opt_dim.value());
+    return at::scatter(self, opt_dim.value(), new_indices, new_values);
+  } else {
+    TORCH_CHECK(indices.dim() == 1, "when axis=None, `indices` must have a single dimension");
+    auto result = at::scatter(self.flatten(), 0, indices, values);
```

Comment:
Actually, can we use  https://pytorch.org/docs/stable/generated/torch.Tensor.put_.html here to avoid calling `flatten` (sparing us a copy)?

Reply:
fixes in [3f06685](https://github.com/pytorch/pytorch/pull/125601/commits/3f06685be5088d1405d507da3c242f0f11e5bd66)